satpy/readers/seviri_l1b_native.py

```diff
@@ -480,6 +516,46 @@ def get_dataset(self, dataset_id, dataset_info):
 
         return dataset
 
+    def pad_hrv_data(self, dataset):
+        """Pad HRV data with empty pixels."""
+        logger.debug('Padding HRV data to full disk')
+        nlines = int(self.mda['hrv_number_of_lines'])
+        nlines_pad = 11136
+        ncols_pad = 11136
+        HRV_bounds = self.trailer['15TRAILER']['ImageProductionStats']['ActualL15CoverageHRV']
+
+        data_pad = np.zeros((nlines_pad, ncols_pad), dtype=dataset.dtype)
+        if np.issubdtype(data_pad.dtype, np.floating):
+            data_pad = data_pad * np.nan
+
+        # lower HRV window
+        lower_south_line = HRV_bounds['LowerSouthLineActual']
+        lower_north_line = HRV_bounds['LowerNorthLineActual']
+        lower_east_col = HRV_bounds['LowerEastColumnActual']
+        lower_west_col = HRV_bounds['LowerWestColumnActual']
+
+        if lower_north_line > lower_south_line:  # we have some of the lower HRV window
+            if lower_west_col - lower_east_col != dataset.data.shape[1] - 1:
+                raise IndexError('East and west bounds of lower HRV window do not match expected data shape')
+
+            data_pad[lower_south_line - 1:lower_north_line, lower_east_col - 1:lower_west_col] = \
+                dataset[lower_south_line - nlines_pad + nlines - 1: lower_north_line - nlines_pad + nlines, :].data
+
+        # upper HRV window
+        upper_north_line = HRV_bounds['UpperNorthLineActual']
+        upper_south_line = HRV_bounds['UpperSouthLineActual']
+        upper_east_col = HRV_bounds['UpperEastColumnActual']
+        upper_west_col = HRV_bounds['UpperWestColumnActual']
+
+        if upper_north_line > upper_south_line:  # we have some of the upper HRV window
+            if upper_west_col - upper_east_col != dataset.data.shape[1] - 1:
+                raise IndexError('East and west bounds of upper HRV window do not match expected data shape')
+
+            data_pad[upper_south_line - 1:upper_north_line, upper_east_col - 1:upper_west_col] = \
+                dataset[upper_south_line - nlines_pad + nlines - 1: upper_north_line - nlines_pad + nlines, :].data
```

Comment:
This is twice the same code basically. Could you extract a new method out of this and use it twice instead?

Reply:
The initial implementation did in fact not pad region of interest (ROI) data. This has now been implemented with the latest commits. Since the information about the lines and columns of the data boundaries is stored differently for the ROI data, this lead to quite a different implementation. To do this, I added a function that checks whether we're dealing with standard FES/RSS data or ROI data (self.is_roi()). In case of standard FES/RSS data, the relevant lines and columns are read from the trailer, whereas for ROI data this information is read from the secondary header. As a result of the many changes, the code segment above has changed considerable, but in general I have tried to remove all instances of duplicate code and make it as generic as possible.

I would be happy to discuss any questions or possible improvements.

Reply:
Sounds good! I'll have another look :)